src/accelerate/utils/dataclasses.py

```diff
@@ -495,7 +495,10 @@ class DeepSpeedPlugin:
         },
     )
     gradient_accumulation_steps: int = field(
-        default=None, metadata={"help": "Number of steps to accumulate gradients before updating optimizer states"}
+        default="auto",
```

Comment:
Hello Zach, Thank you for adding this, but if they specify it via accelerate config/cmd arg, then it won't be updated as the current logic is to check the env variable if this attribute is `None`

Reply:
Will keep as none then! :)

Reply:
(And modify the string as such)